Kinvey.Core/Offline/QueryCacheItem.cs

```diff
@@ -0,0 +1,40 @@
+﻿// Copyright (c) 2018, Kinvey, Inc. All rights reserved.
+//
+// This software is licensed to you under the Kinvey terms of service located at
+// http://www.kinvey.com/terms-of-use. By downloading, accessing and/or using this
+// software, you hereby accept such terms of service  (and any agreement referenced
+// therein) and agree that you have read, understand and agree to be bound by such
+// terms of service and are of legal age to agree to such terms with Kinvey.
+//
+// This software contains valuable confidential and proprietary information of
+// KINVEY, INC and is subject to applicable licensing agreements.
+// Unauthorized reproduction, transmission or distribution of this file and its
+// contents is a violation of applicable laws.
+
+using System;
+using SQLite.Net.Attributes;
+using Newtonsoft.Json;
+
+namespace Kinvey
+{
+    public class QueryCacheItem
+    {
+        [PrimaryKey, AutoIncrement]
+        public int key { get; set; }
+
+        public string collectionName { get; set; }
+
+        public string query { get; set; }
```

Comment:
@vinaygahlawat wondering if there's a way to create a compound unique constraint using `collectionName` and `query`

Reply:
That is a good question. Would be nice to have a composite key for this table, but right now it is not supported by the underlying package.

https://github.com/oysteinkrog/SQLite.Net-PCL/issues/72